packages/sanity/test/descriptors/utils.ts

```diff
@@ -0,0 +1,218 @@
+
+  // Handle arrays
+  if (Array.isArray(o)) {
+    const filtered = o
+      .map((item) => deleteUnserializableTypes(item))
+      .filter((item) => item !== undefined)
+
+    return filtered.length > 0 ? filtered : undefined
+  }
+
+  // Handle objects
+  for (const key of Object.keys(o)) {
+    const value = o[key]
+
+    if (value === null) {
+      // Keep null values if you want, or delete them
+      continue
+    }
+
+    if (typeof value === 'object') {
+      // Check for unserializable types
+      if (value.__type === 'function' || value.__type === 'cyclic' || value.__type === 'maxDepth') {
+        delete o[key]
+      } else {
+        // Recursively clean nested objects/arrays
+        const cleaned = deleteUnserializableTypes(value)
+
+        if (cleaned === undefined) {
+          delete o[key]
+        } else if (
+          typeof cleaned === 'object' &&
+          !Array.isArray(cleaned) &&
+          Object.keys(cleaned).length === 0
+        ) {
+          // Delete empty objects
+          delete o[key]
+        } else {
+          o[key] = cleaned
+        }
+      }
+    }
+  }
+
+  // Return undefined if the object is now empty
+  if (Object.keys(o).length === 0) {
+    return undefined
+  }
+
+  return o
+}
+
+function onlyNoopFieldsets(fieldsets: any[], groups: any[]) {
+  // extractWorkspaceManifest excludes fieldsets which are singluar (.single == true) except for when there are groups
+  if (fieldsets.length === 1 && (groups || []).length <= 1) {
+    return true
+  }
+
+  return fieldsets.every(
+    (fieldset) =>
+      Object.values(fieldset).filter((v) => {
+        return v !== undefined
+      }).length <= 1,
+  )
+}
+
+export function normalizeConvertedType(obj: any) {
+  if (!obj) return obj
+
+  // Titles are not inherited
+  if (obj.extends === 'string' && obj.title === 'String') {
+    obj.title = undefined
+  }
+
+  if (obj.extends === 'block' && obj.title === 'Block') {
+    obj.title = undefined
+  }
+
+  if (obj.extends === 'text' && obj.title === 'Text') {
+    obj.title = undefined
+  }
+
+  if (obj.options && Object.keys(obj.options).length === 0) {
+    obj.options = undefined
+  }
+
+  // Apply recursive normalization to nested structures
+  if (Array.isArray(obj.of)) {
+    for (const item of obj.of) {
+      if (item && typeof item === 'object') {
+        normalizeConvertedType(item.typeDef)
+      }
+    }
+  }
+
+  if (Array.isArray(obj.fields)) {
+    for (const field of obj.fields) {
+      if (field && typeof field === 'object') {
+        normalizeConvertedType(field.typeDef)
+      }
+    }
+  }
+
+  return obj
+}
+
+export function normalizeSchemaType(obj: any, nested = false) {
+  if (!obj) return obj
+
+  // Titles are not inherited on nested types (i.e. fields)
+  if (nested && obj.extends === 'block' && obj.title === 'Block') {
+    obj.title = undefined
+  }
+
+  if (nested && obj.extends === 'string' && obj.title === 'String') {
+    obj.title = undefined
+  }
```

Comment:
Are these actually needed? I thought the built-in schemas didn't set these `title`? 🤔